rules/xcodeproj.bzl

```diff
@@ -280,28 +280,57 @@ def _gather_asset_sources(target_info, path_prefix):
         https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#target-source
     """
     asset_sources = []
-
-    # Each key will be a short_path to a directory ending either .xcdatamodeld or .xcdatamodel
     datamodel_groups = {}
+    catalog_groups = {}
+
     for s in target_info.asset_srcs.to_list():
         short_path = s.short_path
         group = paths.dirname(short_path)
+        short_path_components = short_path.split("/")
+        is_xcassets = False
```

Comment:
instead of 5 different boolean variables,  maybe its simpler to store the extension? also, probably makes sense to extract this out into a `_classify_asset` helper function?